### PR TITLE
Improved find path

### PR DESCRIPTION
**Pull request by TeachingGame**
Updated Find path for user to specify how many consecutive wins they would like to have before the bot stops and alerts them to a potential good seed.
The code seems to work fine except the discord message failed on my attempt (did my net drop or is it a code error). I will recheck tomorrow but thought i'd post here in case I missed something.
the error
22:50:02 | INFO: Congratulations!
22:50:02 | DEBUG: Saving a screenshot to logs\cresselia_2021-03-25 22-33-00_cap_1.png
22:50:14 | ERROR: The Discord webhook failed to send: ('Connection aborted.', ConnectionResetError(10054, 'An existing connection was forcibly closed by the remote host', None, 10054, None))
22:50:14 | INFO: This path won 1 times with 3 lives remaining.

### CODE REVIEW

File: Config.sample.toml [hunk 1]
```diff
@@ -57,10 +57,13 @@ LEGENDARY_BALLS = 500
 #   "KEEP PATH" - Will force the game to always reset on a pre-saved path.
 #       Similar to "BALL SAVER" and "STRONG BOSS" but will always use the same path if it fails.
 #       This will deplete your Ore quickly.
-#   "FIND PATH" - Will run like default mode but will stop as soon as it win once. It can be used
-#       alongside KEEP PATH to find a winning path. It's up  to the user to check if the path was
-#       good or not.
+#   "FIND PATH" - Will run like default mode but will stop when it consecutively wins the number of
```
Comment (pifopi) [inline Config.sample.toml]:
```suggestion
#   "FIND PATH" - Will run like STRONG BOSS mode but will stop when it consecutively wins the number of 
```

File: Config.sample.toml [hunk 2]
```diff
@@ -57,10 +57,13 @@ LEGENDARY_BALLS = 500
 #   "KEEP PATH" - Will force the game to always reset on a pre-saved path.
 #       Similar to "BALL SAVER" and "STRONG BOSS" but will always use the same path if it fails.
 #       This will deplete your Ore quickly.
-#   "FIND PATH" - Will run like default mode but will stop as soon as it win once. It can be used
-#       alongside KEEP PATH to find a winning path. It's up  to the user to check if the path was
-#       good or not.
+#   "FIND PATH" - Will run like default mode but will stop when it consecutively wins the number of 
+#       times inputted into "FIND PATH WINS".
+#       It can be used alongside KEEP PATH to find a winning path. Don't forget to update the  
+#       CONSECUTIVE WINS section below, before starting KEEP PATH.
```
Comment (pifopi) [inline Config.sample.toml]:
```suggestion
#       CONSECUTIVE_RESETS section below, before starting KEEP PATH.
```

**Review by pifopi — APPROVED** [review]
It can be tricky for users to understand what this mode is doing for many reasons (listed below) but I'm totally fine with that, Find path is a "beta" mode

- One reset doesn't mean we just won. For example if you find a non legendary pokemon you want to hunt for, you can reset while you lost
- Not resetting doesn't mean you lost, for example, you could be lacking ore to do that. In that case, they'll get an infinite loop
- Users might lose a good path because it lost due to bad RNG

**Comment by TeachingGame** [comment]
> It can be tricky for users to understand what this mode is doing for many reasons (listed below) but I'm totally fine with that, Find path is a "beta" mode
> 
> * One reset doesn't mean we just won. For example if you find a non legendary pokemon you want to hunt for, you can reset while you lost
> * Not resetting doesn't mean you lost, for example, you could be lacking ore to do that. In that case, they'll get an infinite loop
> * Users might lose a good path because it lost due to bad RNG

Thank you for the review and the comment. I was trying to minimise adding new code so decided to use the already existing CONSECUTIVE RESETS variable but didn't take into account some of your suggestions.
I will add an additional warning not to use NON_LEGEND hunting at the same time as it can produce false positives.
I could add another counter to track actual CONSECUTIVE_WINS but it might be cleaner to leave it the way it is. It is a beta mode as you suggested.

**Pull request merged**